src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -1287,102 +1306,122 @@ bool emberAfThermostatClusterSetActivePresetRequestCallback(
     return true;
 }
 
-bool emberAfThermostatClusterStartPresetsSchedulesEditRequestCallback(
-    chip::app::CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
-    const chip::app::Clusters::Thermostat::Commands::StartPresetsSchedulesEditRequest::DecodableType & commandData)
+bool validAtomicAttributes(const chip::app::Clusters::Thermostat::Commands::AtomicRequest::DecodableType & commandData)
 {
-    ScopedNodeId sourceNodeId = GetSourceScopedNodeId(commandObj);
-
-    EndpointId endpoint = commandPath.mEndpointId;
-
-    // If the presets are editable and the scoped node id of the client sending StartPresetsSchedulesEditRequest command
-    // is not the same as the one that previously originated a StartPresetsSchedulesEditRequest command, return BUSY.
-    if (gThermostatAttrAccess.GetPresetsEditable(endpoint) &&
-        (gThermostatAttrAccess.GetOriginatorScopedNodeId(endpoint) != sourceNodeId))
+    auto attributeIdsIter = commandData.attributeRequests.begin();
+    bool requestedPresets, requestedSchedules;
+    while (attributeIdsIter.Next())
     {
-        commandObj->AddStatus(commandPath, imcode::Busy);
-        return true;
-    }
+        auto & attributeId = attributeIdsIter.GetValue();
 
-    // If presets are editable and the scoped node id of the client sending StartPresetsSchedulesEditRequest command
-    // is the same as the one that previously originated a StartPresetsSchedulesEditRequest command, extend the timer.
-    if (gThermostatAttrAccess.GetPresetsEditable(endpoint))
-    {
-        ExtendTimer(endpoint, commandData.timeoutSeconds);
-        commandObj->AddStatus(commandPath, imcode::Success);
-        return true;
+        switch (attributeId)
+        {
+        case Presets::Id:
+            if (requestedPresets)
+            {
+                return false;
+            }
+            requestedPresets = true;
+            break;
+        case Schedules::Id:
+            if (requestedSchedules)
+            {
+                return false;
+            }
+            requestedSchedules = true;
+            break;
+        default:
+            return false;
+        }
     }
-
-    // Set presets editable to true and the scoped originator node id to the source scoped node id, and start a timer with the
-    // timeout in seconds passed in the command args. Return success.
-    gThermostatAttrAccess.SetPresetsEditable(endpoint, true);
-    gThermostatAttrAccess.SetOriginatorScopedNodeId(endpoint, sourceNodeId);
-    ScheduleTimer(endpoint, commandData.timeoutSeconds);
-    commandObj->AddStatus(commandPath, imcode::Success);
     return true;
```

Comment:
Needs to check `attributeIdsIter.GetStatus()`, no?

Also spec says to INVALID_COMMAND if the list is empty, but I don't see that here.